presto-native-sidecar-plugin/src/main/java/com/facebook/presto/sidecar/NoOpPlugin.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sidecar;
+
+import com.facebook.presto.spi.Plugin;
+
+/**
+ * Todo: Remove this class when support for CoordinatorPlugin is added in presto-maven-plugin.
+ */
+public class NoOpPlugin
```

Comment:
Can't we just manually check in the `META-INF` file?

Reply:
`PrestoServer` does not find the `NativeSidecarPlugin` when I try to manually check in the `META-INF` file. 
The one way it works is packaging the `presto-native-sidecar-plugin ` module as `presto-plugin`. In that case, it requires an implementation of `Plugin.class` in the module.

Reply:
Adding to the above discussion:

There are two ways to load the plugins:

1. Using the directory path (`../presto-native-sidecar-plugin/target`)
2. Using the `pom.xml` (`../presto-native-sidecar-plugin/pom.xml`)

All plugins implementing the `Plugin.class` are loaded via the `pom.xml` approach. We can do the same for `CoordinatorPlugin`, but there’s a slight complication.

In the `PluginDiscovery` class (see [this link](https://github.com/prestodb/presto/blob/26fd8dbd976e2b4341558c953e54bcfbad0a6198/presto-main/src/main/java/com/facebook/presto/server/PluginDiscovery.java#L50)), there's a check that ensures the artifact has the `presto-plugin` extension. This check fails because the new `presto-native-sidecar-plugin` module is not packaged as a `presto-plugin`, and hence doesn’t have the required `presto-plugin` extension.

If we were to package the `presto-native-sidecar-plugin` as a `presto-plugin`, it would require an implementation of `Plugin.class`, as specified [here](https://github.com/prestodb/presto-maven-plugin/blob/master/src/main/java/com/facebook/presto/maven/ServiceDescriptorGenerator.java#L82).

A potential fix would be to update the [presto-maven-plugin](https://github.com/prestodb/presto-maven-plugin/tree/master) repository to support packaging of the CoordinatorPlugin as a `presto-plugin`. Specifically, we should allow a module to be packaged as a `presto-plugin` if it contains either `Plugin.class` or `CoordinatorPlugin` implementations. Currently, the packaging fails if it doesn’t detect a `Plugin.class` implementation. Hence, the need for a `NoOpPlugin` class here.

Reply:
Can you create an issue in the Presto Maven plugin repo?

Reply:
Opened an issue: https://github.com/prestodb/presto-maven-plugin/issues/20